designs/sam-debugging/local-sam-debugging.md

```diff
@@ -0,0 +1,308 @@
+
+| Property                | Description                                          | Used by Standalone Lambda Handler | Used by SAM Template Resources |
+| ----------------------- | ---------------------------------------------------- | --------------------------------- | ------------------------------ |
+| SAM Template            | Path to SAM Template file                            |                                   | x                              |
+| SAM Template Resource   | Name of resource within SAM Template                 |                                   | x                              |
+| SAM Template Parameters | Values to use for SAM Template Parameters            |                                   | x                              |
+| Environment Variables   | Environment Variables exposed to the Lambda Function | x                                 | x                              |
+| Input Event             | Payload passed to the invoked Lambda Function        | x                                 | x                              |
+| Runtime                 | Runtime of Lambda Function to invoke                 | x                                 |                                |
+| Handler                 | Lambda Function Handler to invoke                    | x                                 |                                |
+| Timeout                 | Timeout threshold for Lambda function                | x                                 |                                |
+| Memory                  | Memory provided to Lambda function                   | x                                 |                                |
+
+The following SAM CLI related arguments are relevant to debugging both standalone lambda function handlers and sam template resources. For reference see the [sam build](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-build.html) command.
+
+| Property                                                         | Default Value                    |
+| ---------------------------------------------------------------- | -------------------------------- |
+| Build SAM App in container                                       | false                            |
+| Skip new image check                                             | false                            |
+| use a docker network                                             | empty string (no docker network) |
+| additional build args (passed along to `sam build` calls)        | empty string                     |
+| additional local invoke args (passed along to `sam local` calls) | empty string                     |
+
+The following AWS related arguments are relevant to debugging both standalone lambda function handlers and sam template resources:
+
+-   Credentials
+-   Region
```

Comment:
While coming up with the design you questioned why these were required if it's local - might be worth explaining the use-case (e.g. they're injected into the local Lambda container)